Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -6458,6 +6695,907 @@ def get_list_asset_group_command(
+    """Creates a new report configuration from a template."""
+    endpoint = "/api/3/reports"
+
+    payload = {
+        "format": "sql-query",
+        "name": f"XSOAR_Asset_Collector_{int(time.time())}",
+        "query": BASE_QUERY.get(event_type),
+        "version": "2.3.0",
+    }
+
+    demisto.debug(f"Prepared the payload for {event_type}")
+    # Use the client's http_request method for POST
+    response = await client.http_request("POST", endpoint, payload=payload)
+
+    # Handle response logic
+    report_url = response.headers.get("Location")
+    await response.release()
+    if not report_url:
+        raise Exception("Failed to retrieve report URL from Location header.")
+
+    report_id = report_url.split("/")[-1]
+    demisto.debug(f"Report configuration for {event_type} created successfully. Report ID: {report_id}")
+    return report_id
+
+
+async def generate_report(client: InsightVMClient, report_id: str) -> str:
+    """Triggers the generation of a report."""
+    endpoint = f"/api/3/reports/{report_id}/generate"
+
+    # Use the client's http_request method for POST
+    response = await client.http_request("POST", endpoint, payload={})
+
+    # Handle response logic
+    data = await response.json()
+    await response.release()
+    instance_id = str(data.get("id"))
+    demisto.debug(f"Report generation started. Instance ID: {instance_id}")
+    return instance_id
+
+
+async def check_status_of_report(client: InsightVMClient, report_id: str, instance_id: str) -> str:
+    """Checks the status of a report instance with non-blocking waits."""
+    while True:
+        endpoint = f"/api/3/reports/{report_id}/history/{instance_id}"
+
+        # Use the client's http_request method for GET
+        response = await client.http_request("GET", endpoint)
+
+        status_data = await response.json()
+        await response.release()
+        status = status_data.get("status", "unknown")
+
+        demisto.debug(f"Current report status: {status} for {report_id=} and {instance_id=}")
+
+        if status == "complete":
+            return instance_id
+
+        if status in ["failed", "aborted"]:
+            demisto.debug(f"Report status is '{status}'. Re-triggering report generation")
+            instance_id = await generate_report(client, report_id)
+
+        if status in ["generated", "running", "unknown"]:
+            demisto.debug("Report still processing/unknown. Waiting 120 seconds")
+
+        await asyncio.sleep(120)
+
+
+async def stream_report(client: "InsightVMClient", report_id: str, instance_id: str) -> AsyncGenerator[str, None]:
+    """
+    Asynchronously streams the content of a Rapid7 InsightVM report.
+    Yields decoded text lines (strings) to be consumed by the parser.
+    """
+    endpoint = f"/api/3/reports/{report_id}/history/{instance_id}/output"
+    demisto.debug(f"Starting report download stream from: {endpoint}")
```

Comment:
Why are we downloading the report?
The whole issue here is that we can't expect the size of the report, a too-large file will be problematic to download.